akka-stream/src/main/boilerplate/akka/stream/scaladsl/UnzipWithApply.scala.template

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2014-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.impl.GenJunctions._
+import akka.stream.impl.StreamLayout
+
+trait UnzipWithApply {
+
+  [2..20#/**
+   * Create a new `UnzipWith` specialized for 1 outputs.
+   *
+   * @param unzipper unzipping-function from the input value to 1 output values
+   */
+  def apply[O, [#A1#]](unzipper: O ⇒ ([#A1#])): UnzipWith1[O, [#A1#]] = {
+    val shape = new FanOutShape1[O, [#A1#]]("UnzipWith1")
+    new UnzipWith1(shape, new UnzipWith1Module(shape, unzipper, OperationAttributes.name("UnzipWith1")))
+    }
+    #
+
+  ]
```

Comment:
Oh, thanks for the pointer, stupid me: this won’t work! Is this really only a warning?

So, we need to somehow disambiguate the method signatures as seen by the JVM, and the only two ways I can see are:

* using different method names
* not having overloads at all
* (adding dummy arguments with defaults—not really)

The second case would need a dependent return type calculated from the input, we have that in the HTTP routing DSL already. Other suggestions?